File: ServerProject/src/main/java/com/example/meditime/model/MedicationLog.java
//Amy Wickham 121785021
package com.example.meditime.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;
/**
 * Entity class representing a log entry for medication administration.
 * Tracks the scheduled and actual times the medication was given, the status of administration,
 * notes, and the carer responsible for administering the medication.
 */
@Entity
public class MedicationLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long logId;

    @ManyToOne
    private ClientMedication clientMedication;

    @ManyToOne
    //@JoinColumn(name = "carer_id")  // optional: maps to DB column name
    private User carer;  // this is the "carerId" from the diagram

    private String scheduledTime;
    private String actualTime;

    @Enumerated(EnumType.STRING)
    private Status status;

    private String notes;

    public enum Status {
        Given, Skipped, Missed, Late
    }

    // Getters and Setters

    public Long getLogId() {
        return logId;
    }

    public void setLogId(Long logId) {
        this.logId = logId;
    }

    public ClientMedication getClientMedication() {
        return clientMedication;
    }

    public void setClientMedication(ClientMedication clientMedication) {
        this.clientMedication = clientMedication;
    }

    public User getCarer() {
        return carer;
    }

    public void setCarer(User carer) {
        this.carer = carer;
    }

    public String getScheduledTime() {
        return scheduledTime;
    }

    public void setScheduledTime(String scheduledTime) {
        this.scheduledTime = scheduledTime;
    }

    public String getActualTime() {
        return actualTime;
    }

    public void setActualTime(String actualTime) {
        this.actualTime = actualTime;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public String getNotes() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }
}

File: Mobile/app/src/main/java/com/example/medicationapp/model/MedicationLog.kt


package com.example.medicationapp.model

import java.time.LocalDateTime
import java.time.LocalTime

/**
 * Entity class representing a log entry for medication administration in the Room database.
 * Tracks when a medication was scheduled versus when it was actually given, who administered it, and its status.
 * Linked to both ClientMedication and User (carer) entities via foreign key constraints.
 * Includes optional notes and a status enum to indicate the outcome of the medication event.
 * Uses indices on foreign keys for optimized querying.
 */


data class MedicationLog(
    val logId: Long,

    val clientMedicationId: Long,
    val carerId: Long,

    val scheduledTime: List<LocalTime>,
    val actualTime: LocalTime? = null,

    val status: Status,
    val notes: String? = null
) {
    enum class Status {
        Given, Skipped, Missed, Late
    }
}
